src/producer.cc

```diff
@@ -415,7 +415,7 @@ ProducerMessage::ProducerMessage(v8::Local<v8::Object> obj, Topic * topic):
   m_is_empty(true) {
   // We have this bad boy now
 
-  m_partition = GetParameter<int64_t>(obj, "partition", 0);
+  m_partition = GetParameter<int64_t>(obj, "partition", -1);
```

Comment:
Perhaps better to use the constant: http://docs.confluent.io/2.0.0/clients/librdkafka/classRdKafka_1_1Topic.html#ab69c18779df1c8e150648e640b159654